classfile/reader.ts

```diff
@@ -0,0 +1,121 @@
+module J2ME {
+  declare var util;
+  export class Reader {
+    view: DataView;
+    // DataView is not optimized, use Uint8Array for the fast paths.
+    u8: Uint8Array;
+    offset: number;
+    static T: string [] = new Array(124);
+    constructor(buffer: ArrayBuffer, offset: number) {
+      this.view = new DataView(buffer);
+      this.u8 = new Uint8Array(buffer);
+      this.offset = offset || 0;
+    }
+
+    read8() {
+      return this.u8[this.offset++];
+    }
+
+    read16() {
+      var u8 = this.u8;
+      var o = this.offset;
+      this.offset += 2;
+      return u8[o] << 8 | u8[o + 1];
+    }
+
+    read32() {
+      return this.readInteger() >>> 0;
+    }
+
+    readInteger() {
+      var o = this.offset;
+      var u8 = this.u8;
+      var a = u8[o + 0];
+      var b = u8[o + 1];
+      var c = u8[o + 2];
+      var d = u8[o + 3];
+      this.offset = o + 4;
+      return (a << 24) | (b << 16) | (c << 8) | d;
+    }
+
+    readFloat() {
+      var data = this.view.getFloat32(this.offset, false);
+      this.offset += 4;
+      return data;
+    }
+
+    readDouble() {
+      var data = this.view.getFloat64(this.offset, false);
+      this.offset += 8;
+      return data;
+    }
+
+    readStringFast(length: number): string {
```

Comment:
It's similar but it doesn't behave the same. I optimized it because it was showing up in profiles.

1. Avoids temporary string allocation by using a string array and then calling join.
2. It attempts to avoid recompilation in the JS engine caused by setting `array.length`. 
3. It's optimized for use in the reader rather than from an array view.

We could change the one in utils to use this somehow.